lib/logsource.c

```diff
@@ -507,61 +507,59 @@ _unregister_counters(LogSource *self)
 static void
 _allocate_counter_keys(LogSource *self)
 {
-  stats_cluster_key_builder_push(self->metrics.stats_kb);
+  StatsClusterKeyBuilder *kb = self->metrics.stats_kb;
+
+  stats_cluster_key_builder_add_label(kb, stats_cluster_label("id", self->stats_id));
```

Comment:
Won't we add the same label twice in case of a config revert? (init -> deinit -> init)